lib/network/rpc/level-net.js

```diff
@@ -1,343 +1,109 @@
- * @param {Object} options.logger logger object
- * @param {Number} [options.streamMaxPendingAck] max number of
- *   in-flight output stream packets sent to the server without an ack
- *   received yet
- * @param {Number} [options.streamAckTimeoutMs] timeout for receiving
- *   an ack after an output stream packet is sent to the server
- * @return {Object} a server object, not yet listening on a TCP port
- * (you must call listen(port) on the returned object)
- */
-module.exports.createServer = function LevelServer(db, options) {
-    assert(options.logger);
-
-    const httpServer = http.createServer();
-    const ioServer = io(httpServer);
-    const log = options.logger;
-
-    ioServer.initMetadataService = function initMetadataService(baseNs = '') {
-        assert(baseNs === '' || baseNs.startsWith('/'));
-
-        // all metadata operations executed by leveldb go through this
-        // namespace, other operations (ping etc.) go through /misc.
-        const mdSock = this.of(`${baseNs}/metadata`);
-        mdSock.on('connection', conn => {
-            /**
-             * lookup a sublevel db given by the @a path array from
-             * the main db handle.
-             *
-             * @param {String []} path path to the sublevel, as a
-             * piecewise array of sub-levels
-             * @return {Object} the handle to the sublevel
-             */
-            function lookupSubLevel(path) {
-                let subDb = db;
-                path.forEach(pathItem => {
-                    subDb = subDb.sublevel(pathItem);
-                });
-                return subDb;
-            }
-
-            const mdStreams = sioStream.createSocket(
-                conn,
-                options.logger,
-                options.streamMaxPendingAck,
-                options.streamAckTimeoutMs);
-            conn.on('error', err => {
-                log.error('error on socket.io /metadata connection',
-                          { error: err });
-            });
-            conn.on('call', (remoteCall, args, cb) => {
-                // DB commands expect a 'subLevel' property as the
-                // path to the sublevel DB (array of path items).
-                const decodedArgs = mdStreams.decodeStreams(args);
-                if (dbAsyncCommandNames.includes(remoteCall)) {
-                    try {
-                        // We're pushing the callback to arguments
-                        // that async DB sublevel calls expect as
-                        // their last argument.
-                        decodedArgs.rpcArgs.push((err, data) => {
-                            cb(flattenError(err),
-                               mdStreams.encodeStreams(data));
-                        });
+    openSub(subName) {
+        const subDbClient = new LevelDbClient({ url: this.url,
+                                                logger: this.logger });
+        Object.assign(subDbClient, this);
```

Comment:
Yes, we need this because we have to provide the original exposed RPC calls to the new object.